### ISSUE TITLE
Dynamic errorStrategy is not evaluated correctly 

### ISSUE THREAD

**Issue by huguesfontenelle** [nextflow-io/nextflow#183]
I'd like to use a special error strategy for one process: retry if the fails the first time, ignore it if fails again.
That should be possible with the following code:
```
errorStrategy { task.attempt==1 ? 'retry' : 'ignore' }
```

Unfortunately I get:
```
N E X T F L O W  ~  version 0.20.0
Launching no-retry.nf
[warm up] executor > local
[a6/945ef1] Submitted process > rerun (1)
WARN: Error running process > Process 'rerun (1)' terminated with an error exit status -- error is ignored
```
Also the work folder contains only one subfolder/run for that process (instead of the two that I was expecting).

The minimal amount of code (that I could come up with) reproducing the behaviour is:
```

ch = Channel.from(1)

process rerun {
    echo true
    errorStrategy {task.attempt==1 ? 'retry' : 'ignore'}

    input:
    val(no) from ch

    script:
"""
echo "rerun ${no}: ${task.attempt}"
sleep 1
smurf
"""
}
```

**Comment by huguesfontenelle** [comment]
[.nextflow.log](http://pastebin.com/fPDD5CCc)

**Comment by huguesfontenelle** [comment]
I haven't read the code, but I'm betting that process directives are only evaluated once at the beginning of the first run.
Consider this silly example:
```
ch = Channel.from(1)

process rerun {
    echo true
    errorStrategy 'retry'
    maxErrors {task.attempt + 1}

    input:
    val(no) from ch

    script:
"""
echo "rerun ${no}: ${task.attempt}"
sleep 1
smurf
"""
}
```
If `maxErrors` was re-evaluated, this pipeline would never finish. But it does, after two executions of the `rerun` process.

**Comment by pditommaso** [comment]
I've uploaded version `0.20.1-SNAPSHOT` should solve the problem. Give it a try with 

    NXF_VER=0.20.1-SNAPSHOT nextflow run ..

**Comment by pditommaso** [comment]
I've made some other changes. Make sure to try this version: 

```
  Version: 0.20.1-SNAPSHOT build 3835
  Modified: 16-06-2016 19:01 UTC (21:01 CEST)
```

**Comment by pditommaso** [comment]
Fixed with version `0.20.1`

**Comment by odoublewen** [comment]
@pditommaso 

On a somewhat related topic, `task.attempt` seems to be tallied across all instances of a task, rather than the count of attempts for a specific instance, i.e., combination of input values.  I would have expected the latter.

To borrow from @huguesfontenelle 's example -- I would expect the following process to run 12 times (3 tries for each input -- but it only tries 3 times **total**.

Is this the intent?  Is there value in a "per instance" version of these retry settings?

```
ch = Channel.from(1, 2, 3, 4)

process rerun {
    echo true
    errorStrategy 'retry'
    maxErrors { 3 }

    input:
    val(no) from ch

    script:
"""
echo "rerun ${no}: ${task.attempt}"
sleep 1
smurf
"""
}
```
